File: app/src/main/java/com/pinomo/dolmabank/utils/crypto/models/classes/NoCryptoFunction.java
package com.pinomo.dolmabank.utils.crypto.models.classes;

import androidx.annotation.NonNull;

import com.pinomo.dolmabank.utils.crypto.models.abstracts.CryptoFunction;

/**
 * The type No crypto function.
 */
public class NoCryptoFunction extends CryptoFunction<Void> {

    @NonNull
    @Override
    public String crypt(@NonNull String str, boolean isEncrypt) {
        return str;
    }

    @Override
    public Void generateKey() {
        try {
            loadKey(null);
        } catch (Throwable throwable) {
            throwable.printStackTrace();
        }
        return null;
    }

    @Override
    public boolean isValidForKey(Void potentialKey) {
        return true;
    }
}

File: app/src/main/java/com/pinomo/dolmabank/utils/crypto/models/classes/NoCryptoFunction.kt
package com.pinomo.dolmabank.utils.crypto.models.classes

import com.pinomo.dolmabank.utils.crypto.models.abstracts.CryptoFunction

class NoCryptoFunction : CryptoFunction<Unit>() {
    override fun crypt(str: String, isEncrypt: Boolean): String{
        return str
    }

    override fun generateKey() {
        loadKey(Unit)
    }

    override fun isValidForKey(potentialKey: Unit): Boolean {
        return true
    }
}